levels/sre.tex

```diff
@@ -0,0 +1,368 @@
+    \textbf{Product Output}
+    &
+    \textbf{Communication/Writing}
+    &
+    \textbf{Networking}
+    &
+    \textbf{Security}
+    &
+    \textbf{Systems Engineering}
+    &
+    \textbf{Site Reliability Engineering (SRE)}
+    \\
+\hline
+    % Level
+    1
+    &
+
+    % Product Output
+    \cellcolor{blue!25}
+    Creates a design document based on well-defined scoped requirements and
+    implements it.
+    &
+
+    % Communication
+    Reports progress on a regular basis as required by the team's operational
+    requirements. Actively solicits feedback.Participates on interview panels.
+    &
+
+    % Networking
+    Understands common networking issues and troubleshooting techniques.
+
+    \bigbreak
+
+    Understands different networking layers (OSI model).
+
+    \bigbreak
+
+    Understands and can deploy overlay networks with routing and DNS resolution.
+    &
+
+    % Security
+    Understands operating systems security principles like iptables,
+    users/groups, file permissions, and capabilities. Can apply operating system
+    hardening guides like CIS Benchmarks.
+
+    \bigbreak
+
+    Understands basic security principles like SSH keys and TLS certificates.
+    &
+
+    % Systems Engineering
+    &
+
+    % SRE
+    Understands and can make changes to existing Makefiles, shell scripts, and
+    Dockerfiles.
+
+    \bigbreak
+
+    Understands declarative configuration and can use tooling like Terraform or
+    a Kubernetes operator to make changes to existing infrastructure and
+    applications.
+
+    \bigbreak
+
+    May have certification like Certified Kubernetes Administrator (CKA).
+    \\ [13em]
+\hline
+    % Level
+    2
+    &
+
+    % Product Output
+    \cellcolor{blue!25}
+    Can write high quality user and product focused documentation.
+    &
+
+    % Communication
+    Provides constructive review on peers' code and design. Helps new team
+    members during their first weeks.
+    &
+
+    % Networking
+    Can independently troubleshoot most network issues.
+
+    \bigbreak
+
+    Can setup and configure production quality network infrastructure like DNS,
+    Load Balancers, and encrypted networks (like IPsec or WireGuard). Can reason
+    about their reliability.
+    &
+
+    % Security
+    Understands and can apply basic cryptographic principles. Can configure SSH
+    and TLS for a server (chooses appropriate key sizes, algorithms, and versions),
+    pick strong authentication and authorization primitives, and appropriate
+    encryption for data in transit and rest.
+
+    \bigbreak
+
+    Can setup production quality encrypted networking (like IPsec or
+    WireGuard). Can reason about their reliability.
+    &
+
+    % Systems Engineering
+    &
+
+    % SRE
+    Excellent systems troubleshooter.
+
+    \bigbreak
+
+    Familiar with and can configure alerts for system performance (CPU, memory,
+    disk, networking utilization). Uses modern BPF tools for troubleshooting.
```

Comment:
I'd change it to "some tools", as some of them are indeed fairly basic. A lot of these tools would fly waaay over the head of lvl 2 because to _successfully_ use them requires the knowledge of what's underneath, like kernel or syscalls, etc.